optimum/gptq/quantizer.py

```diff
@@ -460,11 +576,7 @@ def store_input_hook(_, input, *args):
                 handle = block.register_forward_pre_hook(store_input_hook, with_kwargs=True)
                 for data in dataset:
                     for k, v in data.items():
-                        # put the data on gpu, we won't put them back to cpu
-                        if not has_device_map or device.type == "cpu":
-                            data[k] = v.to(0)
-                        else:
-                            data[k] = v.to(device)
+                        data[k] = nested_move_to(v, cur_layer_device)
```

Comment:
The failed ci is basically from here. Previous optimum/gptq just moved the cpu model to cuda. The model has no way to run on CPU. It confused users and also explained why the CPU tests require gpu.